File: Aug/Day18/24-game.js
class Solution {
    judgePoint24(cards) {
        const solve = (nums) => {
            if (nums.length === 1) {
                return Math.abs(nums[0] - 24) < 1e-6;
            }
            for (let i = 0; i < nums.length; i++) {
                for (let j = 0; j < nums.length; j++) {
                    if (i !== j) {
                        let remaining = [];
                        for (let k = 0; k < nums.length; k++) {
                            if (k !== i && k !== j) remaining.push(nums[k]);
                        }
                        let a = nums[i], b = nums[j];
                        let candidates = [a + b, a - b, b - a, a * b];

                        for (let val of candidates) {
                            if (solve([...remaining, val])) return true;
                        }
                        if (Math.abs(b) > 1e-6) {
                            if (solve([...remaining, a / b])) return true;
                        }
                        if (Math.abs(a) > 1e-6) {
                            if (solve([...remaining, b / a])) return true;
                        }
                    }
                }
            }
            return false;
        };
        return solve(cards.map(c => c * 1.0));
    }
}

File: Aug/Day18/24-game.java
import java.util.*;

class Solution {
    public boolean judgePoint24(int[] cards) {
        List<Double> nums = new ArrayList<>();
        for (int c : cards) nums.add((double)c);
        return solve(nums);
    }

    private boolean solve(List<Double> nums) {
        if (nums.size() == 1) {
            return Math.abs(nums.get(0) - 24) < 1e-6;
        }
        for (int i = 0; i < nums.size(); i++) {
            for (int j = 0; j < nums.size(); j++) {
                if (i != j) {
                    List<Double> remaining = new ArrayList<>();
                    for (int k = 0; k < nums.size(); k++) {
                        if (k != i && k != j) remaining.add(nums.get(k));
                    }
                    double a = nums.get(i), b = nums.get(j);
                    double[] candidates = {a + b, a - b, b - a, a * b};

                    for (double val : candidates) {
                        List<Double> next = new ArrayList<>(remaining);
                        next.add(val);
                        if (solve(next)) return true;
                    }
                    if (Math.abs(b) > 1e-6) {
                        List<Double> next = new ArrayList<>(remaining);
                        next.add(a / b);
                        if (solve(next)) return true;
                    }
                    if (Math.abs(a) > 1e-6) {
                        List<Double> next = new ArrayList<>(remaining);
                        next.add(b / a);
                        if (solve(next)) return true;
                    }
                }
            }
        }
        return false;
    }
}
